### PR TITLE
Integrate BBjRouter Plugin

### PR DESCRIPTION
**Pull request by DominicWrege**
The PortalFrame takes fully advantage of the [BBjRouter](https://github.com/BBj-Plugins/BBjRouter) Plugin. For each MenuItem, a path based on the caption and tree location will be automatically created. 
It is possible to override the generated path. To keep it simple, all paths are case insensitivity.
Now it is possible to get the path and query parameters, by simple calling the `PortalFrame.getLastRouterEvent` Method.
The Menu will keep a Menu lookup HashMap, containing all registered paths.
Each MenuItem can now have a path parameter placeholder.
If the users types in an unknown route, the PortalFrame will default to a 404 Page.
The 404 Page can be overridden.

### IMPORTANT
The `navigateTo` Method will no longer return a BBjWidget. 
This is now handled  via an event called `PortalFrame.ON_NAVIGATION$`.

### CODE REVIEW

File: framework/PageNotFound/PageNotFound.bbj [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        #this!(window!, window!.getAvailableControlID())
+    methodend
+    
+    method public PageNotFound(BBjWindow wnd!, BBjInt id!)
+         #super!.create(wnd!, id!)
+    methodend
+
+    method public void redraw(Boolean init!)
+        declare BBjChildWindow canvas!
+        if init! > 0 then
```
Comment (ckochaneck) [inline framework/PageNotFound/PageNotFound.bbj:21]: if init! then

File: framework/PageNotFound/PageNotFound.bbj [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            canvas!.addStyle("pageNotFound")
+            ClientUtil.addLocalCSS("WebKit/framework/PageNotFound/PageNotFound.css")
+            statusCode! = UIUtil.addStaticText(canvas!, I18n.getTranslation("404"))
+            statusCode!.addStyle("pageNotFoundStatusCode")
+            #message! =  UIUtil.addStaticText(canvas!, I18n.getTranslation("page-not-found-message"))
+            #message!.addStyle("pageNotFoundMessage")
+        fi
+    methodend
+    
+    method public void setMesage(BBjString msg$)
```
Comment (ckochaneck) [inline framework/PageNotFound/PageNotFound.bbj:32]: Fix Typo

File: framework/PortalFrame/PortalFrame.bbj [hunk 3]
```diff
@@ -193,16 +210,16 @@ REM         footerMenuItem!.setCallback("onChangeTheme")
     
     method public Boolean doModal()
 
-            sysgui! =BBjAPI().openSysGui("X0")
+            #sysgui! =BBjAPI().openSysGui("X0")
```
Comment (ckochaneck) [inline framework/PortalFrame/PortalFrame.bbj:213]: Add spacing after =

File: framework/PageNotFound/PageNotFound.bbj (hunk 2 above)
Comment (DominicWrege) [reply framework/PageNotFound/PageNotFound.bbj:32]: done

File: framework/PageNotFound/PageNotFound.bbj (hunk 1 above)
Comment (DominicWrege) [reply framework/PageNotFound/PageNotFound.bbj:21]: done

**Review by ckochaneck — APPROVED** [review]

**Pull request merged**